bluepyopt/ephys/protocols.py

```diff
@@ -192,7 +193,20 @@ def _run_func(self, cell_model, param_values, sim=None):
             cell_model.freeze(param_values)
             cell_model.instantiate(sim=sim)
 
-            self.instantiate(sim=sim, icell=cell_model.icell)
+            if isinstance(cell_model, models.LFPyCellModel):
+                LFPyCell = cell_model.LFPyCell
+            else:
+                LFPyCell = None
+
+            self.instantiate(
+                sim=sim, icell=cell_model.icell, LFPyCell=LFPyCell
+            )
+
+            if hasattr(sim, "electrode"):
```

Comment:
This looks very hardcoded. Can't you get something based on the type of the sim or cell_model?

Reply:
I agree there is probably a better way to do it. The thing is that one can use the LFPy backend without setting an electrode/probe

Reply:
I'll try to draft a better solution

Reply:
actrually, then we'd move the problem to the `protocol.run()`[https://github.com/alejoe91/BluePyOpt/blob/lfpy/bluepyopt/ephys/protocols.py#L189] function and we would need additional parameters there to be propagated everywhere.

Reply:
With the changes I proposed to the simulator, it could be:
```
           try:
                if isinstance(sim, sim.LFPySimulator):
                    sim.run(
                        lfpy_cell=cell_model.lfpy_cell,
                        lfpy_electrode=cell_model.electrode,
                        tstop=self.total_duration,
                        cvode_active=self.cvode_active)
                else:
                    sim.run(tstop=self.total_duration, cvode_active=self.cvode_active)
```